var/spack/repos/builtin/packages/hpl/package.py

```diff
@@ -101,9 +101,10 @@ def configure(self, spec, prefix):
 
     @when('@2.3:')
     def configure_args(self):
-        config = [
-            'CFLAGS=-O3'
-        ]
+        if '+openmp' in spec:
+            config = ['CFLAGS=-O3 -fopenmp']
```

Comment:
```suggestion
            config = ['CFLAGS=-O3 ' + self.compiler.openmp_flag]
```
This will work regardless of which compiler you are using.